dvc/remote/base.py

```diff
@@ -259,8 +264,9 @@ def get_dir_checksum(self, path_info):
             self.cache.makedirs(new_info.parent)
             self.cache.move(tmp_info, new_info, mode=self.CACHE_MODE)
 
-        self.state.save(path_info, checksum)
-        self.state.save(new_info, checksum)
+        if is_working_tree(self.repo.tree):
+            self.state.save(path_info, checksum)
+        self.state.save(new_info, checksum, tree=self.cache.tree)
```

Comment:
Now that `repo.tree` may be a GitTree, RemoteLocal (`self.cache` here) has it's own WorkingTree instance. When working with GitTree's, `repo.tree` must be used to look for paths in the git-index, and `RemoteLocal.tree` must be used to look for local cache paths. `state.save` also has to know which tree to use - it uses `repo.tree` by default, but when the `repo.tree` is a GitTree, `state.save` must be passed the local cache tree instead